### ISSUE TITLE
Create table name to table id map in zookeeper

### ISSUE THREAD

**Issue by keith-turner** [apache/accumulo#4698]
Tables in zookeeper are conceptually organized as a `Map<TableId, TableName>`.  A lot of code in the Accumulo client needs the inverse `Map< TableName, TableId>`.  There is complex code in the Accumulo client related to inverting the map in zookeeper and keeping it up to date.  If an index were built in zookeeper of the form `Map< TableName, TableId>` then the complex code in the client could possibly be removed.  Also mapping a table name to table id could be more efficient, avoiding always reading all tables to map a single table name to a table id. This would reduce the load on zookeeper server for the case of lots of client and lots of tables.  For more details see the comments on #4684.

**Comment by ctubbsii** [comment]
I think the best way to do this would be to remove all the names from under each table, and create a new single node (maybe per namespace?) that retains the table name mapping for all tables. Otherwise, it would be a nightmare to try to maintain consistency between the locations. I might be interested in working on this or helping direct somebody else. @meatballspaghetti recently asked me for ideas of what issues to work on, and might be interested in this.

**Comment by meatballspaghetti** [comment]
I can look into this.

**Comment by dlmarion** [comment]
@ctubbsii - Can this issue be closed with #4996 merged?

**Comment by ctubbsii** [comment]
> @ctubbsii - Can this issue be closed with #4996 merged?

No, @meatballspaghetti only did the namespace mapping, and my understanding is that it is their intent to finish the table mapping after the chroot work is done, which is close.
